From 230b35c50afb047603898024715f757779a2202a Mon Sep 17 00:00:00 2001
From: Prakash Brahmajyosyula <bprakash@marvell.com>
Date: Tue, 19 Feb 2019 14:33:36 +0530
Subject: [PATCH 1012/1051] octeontx2-af: use blkaddr instead of blktype.

Some variants of Octeontx2 supports multiple blocks of same type.
Correct procdeure to get block is by block address. This patch elimnates
using blktype to get block and fixes issue intorduced in
21ac548cb07f24df2b259ffdd9b13b7d2554b276.

Change-Id: I82e011f3a3fe7434cb2b909d1b1b814a3676c292
Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/4362
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
index 43af6b216c54..a5fef55a8555 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_debugfs.c
@@ -209,7 +209,7 @@ static bool rvu_dbg_is_valid_lf(struct rvu *rvu, int blktype, int lf,
 	}
 
 	hw = rvu->hw;
-	block = &hw->block[blktype];
+	block = &hw->block[blkaddr];
 
 	if (lf < 0 || lf >= block->lf.max) {
 		dev_warn(rvu->dev, "Invalid LF: valid range: 0-%d\n",
-- 
2.17.1

